src/lib/sticky-header/sticky-header.ts

```diff
@@ -46,6 +47,19 @@ const STICK_END_CLASS = 'cdk-sticky-header-end';
  */
 const DEBOUNCE_TIME: number = 5;
 
+export const STICKY_HEADER_SUPPORT_STRATEGY = new InjectionToken('sticky-header-support-strategy');
+
+export function STICKY_HEADER_SUPPORT_STRATEGY_FACTORY() {
+  return isPositionStickySupported();
+}
+
+/** Create a factory for sticky-positioning check to make code more testable */
+export const STICKY_HEADER_SUPPORT_STRATEGY_PROVIDER = {
+  provide: STICKY_HEADER_SUPPORT_STRATEGY,
+  deps: [],
+  useFactory: STICKY_HEADER_SUPPORT_STRATEGY_FACTORY
```

Comment:
You can use `useFactory: isPositionStickySupported`, and move this part to index.ts:
```
providers: [{provide: STICKY_HEADER_SUPPORT_STRATEGY, useFactory: isPositionStickySupported}]
```